Adding setup for China examples

langsmith opened the pull request:
cc @Chaoba @zugaldia @chriswu42  @springmeyer 

This pr creates the structure for adding China examples to this app and adds a simple `ChinaMapView` example as a start 🇨🇳🚀 

It takes advantadge of build flavoring to include firebase in global flavor and exclude firebase in china flavor.

Decisions about UI still need to be made and can be developed in a new pr. Primarily, what sections to show or not show in the nav drawer when the build is a China/`nogpservices` flavor.

zugaldia commented on MapboxAndroidDemo/build.gradle:
```diff
@@ -111,7 +114,8 @@ dependencies {
     // Mapbox dependencies
-    api(dependenciesList.mapboxMapSdk)
+    nogpservicesImplementation dependenciesList.mapboxChinaPlugin
+    gpservicesApi(dependenciesList.mapboxMapSdk)
```
Considering that the Google Play Services-free flavor is now only targeting the China use case (as opposed to non-Play Services stores in general), how about taking this opportunity to rename the two flavors two something clearer, like `china` instead of `nogpservices`? Also, one of the entries include the brackets, while the other doesn't.

langsmith replied:
Fixed

zugaldia commented on MapboxAndroidDemo/src/gpservices/java/com/mapbox/mapboxandroiddemo/MapboxApplication.java:
```diff
@@ -15,7 +15,14 @@ public void onCreate() {
     initializeFirebaseApp();
     setUpPicasso();
-    Mapbox.getInstance(this, getString(R.string.access_token));
+    if (BuildConfig.FLAVOR.matches("gpsservices")) {
```
I think that we'd want to have different folders for the two flavors (and include two versions of this file), instead of using conditional checks, under `src/nogpservices/java/com/mapbox/mapboxandroiddemo/MapboxApplication.java`.

zugaldia commented on MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/MainActivity.java:
```diff
@@ -1143,5 +1153,16 @@ private void initializeModels() {
       null,
       R.string.activity_basic_mapbox_kotlin_url, true, BuildConfig.MIN_SDK_VERSION));
+
+    if (isChinaBuild) {
```
Same as before, instead of using a condition check here, we could have two different versions of `MainActivity.java`.

Chaoba commented:
@zugaldia @langsmith please review this version

langsmith commented on MapboxAndroidDemo/src/main/AndroidManifest.xml:
```diff
@@ -183,8 +183,6 @@
                 android:value="com.mapbox.mapboxandroiddemo.MainActivity" />
         </activity>
 
-        <service android:name=".account.AccountRetrievalService" />
```
We need this service for signing into a Mapbox account. Please add it back into the manifest file.

langsmith replied:
I'll do it

langsmith commented:
thanks for continuing the work on this @Chaoba .  The flavor is working and looking good.

I made some adjustments since your last commit:

1. I adjusted the manifest to re-add the service that you deleted from the manifest. The service is needed for the login process if someone wants to sign into their Mapbox account via the app.
2. I removed the flavor checking from the `china` flavor's `MainActivity` file. The file's only used in the `china` flavor, so no checking is needed.
3. I re-added the `SimpleChinaMapViewActivity` example
4. Removed the `China` nav drawer section. If it's the `china` flavor, the examples will just appear in the same sections that are present in the `global` flavor. For example, `SimpleChinaMapViewActivity` will be in the `Basics` nav drawer section, rather than a `China` nav drawer section.

langsmith commented:
Global flavor working as expected (normal demo app experience):

![ezgif com-resize 1](https://user-images.githubusercontent.com/4394910/47688205-71302b80-dba1-11e8-9a62-388c9e254dea.gif)


China flavor working as expected (simple `ChinaMapView` is the only one visible):

**Can't currently add a GIF because internet is not working at SF office.**

Chaoba commented:
@langsmith @zugaldia Please review this PR and merge it if no other issues, it blocks our works on China demo, thanks.

langsmith commented:
Looks great @Chaoba . Both flavors are working as intended and CI is passing. If you make me a reviewer on this pr, I can ✅

Chaoba commented:
@zugaldia This may affect the global flavor, would you please do the final confirmation?

zugaldia commented:
>  If you make me a reviewer on this pr, I can white_check_mark

I don't think we can add you as a reviewer because you created the PR originally. However, if you're good to go with the current changes, @Chaoba please go ahead merging.

Because these changes are substantial, let's publish (the global flavor) as an alpha/beta release for some time before promoting it to final.

langsmith commented:
> you created the PR originally

Wow. Was a long day.

> let's publish (the global flavor) as an alpha/beta release for some time before promoting it to final

Ok, will do
